schema/elements.go

```diff
@@ -48,6 +50,10 @@ type TypeRef struct {
 	// Either the name or one member of Atom should be set.
 	NamedType *string `yaml:"namedType,omitempty"`
 	Inlined   Atom    `yaml:",inline,omitempty"`
+
+	// If `NamedType` is non-nil and refers to a map-type, this field overrides
```

Comment:
First, I don't think we should limite this to "NamedType".

Second of all, how much have you though of the "override" vs not-override? My knee-jerk reaction would also be that override is the way to go.

Reply:
I am leaning towards it being an override sort of behavior, rather than only applying when the initial is unset. I think that  to not override the original option when a field-level annotation is explicitly applied would be surprising.